app/controllers/application_controller.rb

```diff
@@ -243,10 +243,15 @@ def execute_file_download(study)
                       alert: 'We are unable to process your download.  Please try again later.' and return
         end
       else
-        # send notification to the study owner that file is missing (if notifications turned on)
-        SingleCellMailer.user_download_fail_notification(study, params[:filename]).deliver_now
-        redirect_to merge_default_redirect_params(view_study_path(accession: study.accession, study_name: study.url_safe_name), scpbr: params[:scpbr]),
-                    alert: 'The file you requested is currently not available.  Please contact the study owner if you require access to this file.' and return
+        if requested_file.size.nil?
```

Comment:
why don't we move this check to the beginning of the method and just check `params[:filename].blank?` to be more explicit about what we are testing?

That also fits better into the typical controller pattern of "first validate input, then do stuff, then return"

Reply:
c2bc214